### ISSUE TITLE
Feature request: date picker

### ISSUE THREAD

**Issue by fballiano** [Polymer/paper-elements#20]
Hi, I know that probably this isn't the right place for requests but I think polymer (specifically the paper collection) need an official date/time picker.

a couple of links:
https://github.com/christopheclc/polymer-date-picker
https://groups.google.com/forum/#!topic/polymer-dev/IJ35qtWzlxc

thanks
bye

**Comment by StephanBeutel** [comment]
Great work - but after selecting a date my complete app freezes.

**Comment by RobertLowe** [comment]
I'm a little bit taken by this issue; a date component is necessary for most applications. @David-Mulder has an okay start. The others I've checked out on github aren't ready. 

Something that works for *evergreen* and *mobile devices* would be highly appreciated.

**Comment by AutoSponge** [comment]
I added this with v0.5 for all of the deps and it seems to be working great.  Thanks!

**Comment by ivisedo-byhours** [comment]
I use this component with 0.5 and all work well! 

Only, when I use open() method I need to call manually refreshScrollPosition if value is set.

Congratulations!

**Comment by danieleliberti** [comment]
How can I bind the input field after press OK button with the selected date?
Please create a Simple code jsfiddle or similar.

THX for good your work ,)

**Comment by bendavis78** [comment]
I've created [paper-date-picker](https://github.com/bendavis78/paper-date-picker/) ([docs](http://bendavis78.github.io/paper-date-picker/paper-date-picker/index.html) / [demo](http://bendavis78.github.io/paper-date-picker/paper-date-picker/demo.html)) and [paper-time-picker](https://github.com/bendavis78/paper-time-picker/)  ([docs](http://bendavis78.github.io/paper-time-picker/paper-time-picker/index.html) / [demo](http://bendavis78.github.io/paper-time-picker/paper-time-picker/demo.html))  components with the goal to mirror the android versions as closely as possible, while still having decent performance on mobile devices. I opted against infinite scroll for the date picker, as I couldn't seem to find a way to make it work without sacrificing performance.

Please test on your own devices and let me know what you think (pull requests welcomed)!

**Comment by JaySunSyn** [comment]
great work @bendavis78! Did you try core-list instead of template-repeat performance wise? Are you working on a Polymer 0.9 compatible release?

**Comment by bendavis78** [comment]
@JaySunSyn, yes I worked a regrettably long time trying to make it work with `core-list`.  It's handy for some things, but still buggy and difficult to work with for complex lists. Another big challenge was getting the picker to jump to a specific month. I tried using each month as an item as well as grid mode with each day as an item. I think it's just too many reflows and repaints for a mobile browser. But, I'd love it if someone could find a better way. 

I'll be working on a polymer 0.9 release for both pickers as soon as the dependent elements are working in 0.9.

**Comment by JaySunSyn** [comment]
@bendavis78 I think with Polymer 0.9 we have the simpleness of template-repeat and the performance of core-list (?) in dom-repeat - maybe this will solve the problem ;) Would love to see a 0.9 release. Thanks!

**Comment by bendavis78** [comment]
@JaySunSyn, hmm, it doesn't look to me like `dom-repeat` is a drop-in replacement for `core-list` (unless I'm mistaken). The point of `core-list` is to provide fast scrolling on very large lists by adding and removing dom nodes dynamically while scrolling.

**Comment by JaySunSyn** [comment]
@bendavis78 didn't see a core-list replacement for a while but you are right I've seen iron-list is in progress..

**Comment by rkstar** [comment]
@bendavis78 this is REALLY good!  what about 1.0?  any plans to support it?

**Comment by addyosmani** [comment]
iron-list is very much a work in progress and we're still aiming to get it out. Regarding a 1.0 date picker: I'd recommend collaborating to port over one of the existing 0.5 community pickers to 1.0 or waiting until iron-list is ready if it is a hard dependency.

**Comment by paulobarcelos** [comment]
[iron-list](https://github.com/PolymerElements/iron-list) seems to be fully ported now...

**Comment by kungufli** [comment]
+1

**Comment by FreddyFY** [comment]
+1

**Comment by gertcuykens** [comment]
Please clone bendavis78 stuff in the PolymerElements to prevent multiple developers creating the same code. A material Date / Time picker is a must have basic element. More used then Au Ne Pt elements combined. There is one material design spec so its not that they are going to look very different. And if they do it should be the style only, not the design. Thank you.

**Comment by robertotucci** [comment]
@gertcuykens +1

**Comment by ronnyroeller** [comment]
+1

**Comment by Ioane5** [comment]
+1 please

**Comment by eximius313** [comment]
also "time picker" and "duration picker" are essential components

**Comment by rwestlund** [comment]
+1

**Comment by IftachSadeh** [comment]
+1

**Comment by yordis** [comment]
+1

**Comment by esd100** [comment]
+1

**Comment by danielvanmil** [comment]
+1
